cmd/certgen.go

```diff
@@ -0,0 +1,223 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cilium/certgen/internal/defaults"
+	"github.com/cilium/certgen/internal/generate"
+	"github.com/cilium/certgen/internal/logging"
+	"github.com/cilium/certgen/internal/logging/logfields"
+	"github.com/cilium/certgen/internal/option"
+	"github.com/cilium/certgen/internal/version"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const binaryName = "cilium-certgen"
+
+var log = logging.DefaultLogger.WithField(logfields.LogSubsys, binaryName)
+
+func New() *cobra.Command {
+	vp := viper.New()
+	rootCmd := &cobra.Command{
+		Use:           binaryName + " [flags]",
+		Short:         binaryName,
+		Long:          binaryName + " bootstraps TLS certificates and stores them as K8s secrets",
+		SilenceErrors: true,
+		Version:       version.Version,
+		Run: func(cmd *cobra.Command, args []string) {
+			option.Config.PopulateFrom(vp)
+
+			if option.Config.Debug {
+				logging.DefaultLogger.SetLevel(logrus.DebugLevel)
+			}
+
+			log.Infof("%s %s", binaryName, version.Version)
+
+			if err := generateCertificates(); err != nil {
+				log.WithError(err).Fatal("failed to generate certificates")
+			}
+		},
+	}
+	rootCmd.SetVersionTemplate("{{with .Name}}{{printf \"%s \" .}}{{end}}{{printf \"v%s\" .Version}}\n")
+
+	flags := rootCmd.PersistentFlags()
```

Comment:
nit: there are no sub-commands so these flags to need to be persistent.
```suggestion
	flags := rootCmd.Flags()
```